hs_collection_resource/views.py

```diff
@@ -0,0 +1,103 @@
+                                              needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE,
+                                              raises_exception=True)
+
+        collection_content_res_id_list = []
+        for res_id in request.POST.getlist("resource_id_list"):
+            collection_content_res_id_list.append(res_id)
+
+        if collection_res_obj.metadata.collection:
+            element_id = collection_res_obj.metadata.collection.id
+            hydroshare.resource.update_metadata_element(collection_res_obj.short_id, 'Collection',
+            resource_id_list=collection_content_res_id_list, element_id=element_id)
+        else:
+            hydroshare.resource.create_metadata_element(collection_res_obj.short_id,
+            'Collection', resource_id_list=collection_content_res_id_list)
+
+        current_sharing_status = "Private"
+        if collection_res_obj.raccess.public:
+            current_sharing_status = "Public"
+        elif collection_res_obj.raccess.discoverable:
+            current_sharing_status = "Discoverable"
+
+        new_sharing_status = ""
+        if collection_res_obj.raccess.public or collection_res_obj.raccess.discoverable:
+            downgrade = False
+
+            if len(collection_content_res_id_list) == 0:
+                downgrade = True
+            else:
+                for checked_res_id in collection_content_res_id_list:
+                    res_checked = hydroshare.get_resource_by_shortkey(checked_res_id)
+                    if not res_checked.raccess.public and not res_checked.raccess.discoverable:
+                        downgrade = True
+                        break
+
+            if downgrade:
+                set_resource_flag(request, collection_res_obj.short_id, t="make_private")
+                new_sharing_status = "Private"
+
+        # check if current user is owner or super_user (owner and super_user can change sharing status on frontend)
+        user_permission = "Edit"
+        if collection_res_obj in user.uaccess.owned_resources or user.is_superuser:
+            user_permission = "Own"
+
+        if collection_res_obj.metadata.has_all_required_elements():
+            metadata_status = "Sufficient to make public"
+        else:
+            metadata_status = "Insufficient to make public"
+
+        resource_modified(collection_res_obj, user)
+        ajax_response_data = {'status': 'success', 'user_permission': user_permission, \
+                              'current_sharing_status': current_sharing_status, \
+                              'new_sharing_status': new_sharing_status, 'metadata_status': metadata_status}
+        return JsonResponse(ajax_response_data)
+
+    except Exception as ex:
+        logger.exception("update_collection: %s" % (ex.message))
+        ajax_response_data = {'status': 'error', 'message': ex.message}
+        return JsonResponse(ajax_response_data)
+
```

Comment:
You can just have one return statement here instead of 2 (line#73 and 78) return statements.